### PR TITLE
More general division by triangular matrices

### PR DESCRIPTION
**Pull request by mateuszbaran**
I've added right division by triangular matrices, division of structured matrices and removed some related dead code.

### CODE REVIEW

File: src/triangular.jl [hunk 1]
```diff
@@ -1,11 +1,19 @@
...
+@inline adjoint(A::UnitLowerTriangular{<:Any,<:StaticMatrix}) =
+    UnitUpperTriangular(adjoint(A.data))
+@inline transpose(A::UpperTriangular{<:Any,<:StaticMatrix}) =
+    LowerTriangular(transpose(A.data))
+@inline adjoint(A::UpperTriangular{<:Any,<:StaticMatrix}) =
+    LowerTriangular(adjoint(A.data))
+@inline transpose(A::UnitUpperTriangular{<:Any,<:StaticMatrix}) =
+    UnitLowerTriangular(transpose(A.data))
+@inline adjoint(A::UnitUpperTriangular{<:Any,<:StaticMatrix}) =
+    UnitLowerTriangular(adjoint(A.data))
```
Comment (c42f) [inline src/triangular.jl:16]: It's surprising that Base doesn't seem to include these rules.

**Review by c42f — APPROVED** [review]
As always, I love a good PR which removes code... but this adds functionality as well, very nice :)

File: src/triangular.jl (hunk 1 above)
Comment (mateuszbaran) [reply src/triangular.jl:16]: Is it worth opening an issue in Julia repository?

Comment (c42f) [reply src/triangular.jl:16]: Sure. I can't think of a reason these shouldn't be structure-preserving.

**Pull request merged**